### DISCUSSION TITLE
how can i change the src value of a nextjs Image?

### DISCUSSION THREAD

**Discussion by abedshaaban** [vercel/next.js#46704] [Help] [answered]
### Summary

the following code switches the image src for responsiveness. for example here when the window-width is greater than 600px, the /svg/top.svg is rendered but when the width becomes less than 600px, /svg/top_mobile.svg is rendered. it works fine with `<img>` but not with `next/Image`

```
<picture>
            <source srcSet="/svg/top_mobile.svg" media="(max-width:600px)" />
            
            <Image
              src="/svg/top.svg"
              alt="curve"
              priority
            />
</picture>
```

**Comment by ciruz** [comment, marked as answer]
I don't think what you try with `<picture>` and `next/image` is possible, but I would suggest two other solutions.

**Solution 1:**

With css and media queries, depending on the resolution of the client, show Image 1 or 2, I made a short example with `tailwindcss`:

```javascript
<div className={`block md:hidden`}>
  <Image src="/1.jpg" alt="" width={200} height={200} />
</div>

<div className={`hidden md:block`}>
  <Image src="/2.jpg" alt="" width={200} height={200} />
</div>
```
> `block md:hidden` means: visible on mobile, until `768px`, then it's hidden
> `hidden md:block` means: hidden on mobile, until `768px` then it's visible

A modern Browser would only load the visible image, if you change the resolution, the other image gets loaded, (check this example out and watch your network tab, I tried it on the latest Chrome).

___

**Solution 2:**

With React and `react-responsive`, there is only one image in the DOM.

```jsx
import React, { useEffect, useState } from "react";
import { useMediaQuery } from "react-responsive";

//...

  const [hydrated, setHydrated] = useState(false);

  const isMobile = useMediaQuery(
    {
      query: "(min-width:0px) and (max-width: 768px)",
    },
    hydrated ? undefined : { deviceWidth: 660 }
  );

  useEffect(() => {
    setHydrated(true);
  }, []);

return (
  <div>
    {isMobile ? <Image src="/1.jpg" alt="" width={200} height={200} /> : <Image src="/2.jpg" alt="" width={200} height={200} /> }
  </div>
);
```

**Reply by abedshaaban** [reply]
thanks for your suggested solutions, I took solution 1 and implemented it right away.
